### ISSUE TITLE
Update examples in manual using `jl_`

### ISSUE THREAD

**Issue by omus** [JuliaLang/julia#17107]
While [updating the doctests](https://github.com/JuliaLang/julia/pull/17106) I found that the `jl_` function as mentioned in the [manual](http://julia.readthedocs.io/en/latest/devdocs/types/#typevars) produces different output between 0.4 and 0.5:

Julia 0.4.6
```julia
julia> jl_(x) = ccall(:jl_, Void, (Any,), x)
jl_ (generic function with 1 method)

julia> candid{T}(A::Array{T}, x::T) = 0
candid (generic function with 1 method)

julia> jl_(first(methods(candid)))
Method(sig=Tuple{Array{#T<:Any, N<:Any}, #T<:Any}, va=false, isstaged=false, tvars=#T<:Any, func=#<function>, invokes=nothing, next=nothing)
```

Julia 0.5-dev
```julia
julia> jl_(x) = ccall(:jl_, Void, (Any,), x)
jl_ (generic function with 1 method)

julia> candid{T}(A::Array{T}, x::T) = 0
candid (generic function with 1 method)

julia> jl_(first(methods(candid)))
Main.candid(...)
```

**Comment by yuyichao** [comment]
I believe the change is intentional and this is not a regression. The old printing is almost useless since it prints all the linked methods. It might be useful to print other fields though.

**Comment by omus** [comment]
Ok. Since the change is intentional then someone will need to update the [devdocs type section](http://julia.readthedocs.io/en/latest/devdocs/types/#typevars) that uses `jl_` to show `TypeVar` information.

**Comment by tkelman** [comment]
I think Jeff fixed this one in a8d796355ac99750e7e192e4e31021e1da32cbe9 by adding `.sig`. Do you think that's enough to close this @omus?

**Comment by omus** [comment]
I think the `.sig` solution is good enough.
